blogs/2023/11/13/vscode-copilot-smarter.md

```diff
@@ -0,0 +1,175 @@
+
+In addition, the Command Palette now supports similarity search, so you no longer need to know the exact name of the command when you search for it. You no longer need to speak the unique VS Code jargon to unlock all the goodness the team ships month over month.
+
+This is just the beginning of the `@vscode`` agent. We plan to support more and more scenarios that will allow users to better understand and fully control VS Code.
+
+### Slash commands
+
+Agents can also contribute what we call **slash commands**, which are shortcuts to specific functionality provided by the agent. One of the tasks when answering questions is to determine the intent, understanding what you want to do.
+
+We can infer that "Create a new workspace with Node.js Express Pug TypeScript" means that you want a new project, but `@workspace /new node.js Express Pug TypeScript` is explicit, concise, and saves you time typing.
+
+![New slash command](new-slash-command.png)
+
+Once the intent is clear, the `@workspace` agent has a much better chance of addressing the user's needs, despite the inherent ambiguity of natural language. `@workspace` agent can propose a directory structure and users can click on the proposed files to preview them. There is a **Create Workspace** button that will generate these files in a new folder. This is possible thanks to the `@workspace` agent having access to a tool that can populate the workspace with files.
+
+![@workspace /new results displaying project tree and Create Workspace button](create-workspace-tree-button.png)
+
+## Extensibility
+
+"VS Code is just a shell, you need extensions to make it shine!" – the usual meeting anthem of Microsoft teams working on VS Code extensions, proudly flaunting their role in VS Code's success. We, as the VS Code core team, totally agree with them – VS Code would not be the product it is today without the rich extension ecosystem! AI is no different, and while the Core AI experience lights up with Copilot, our vision is that all the extensions from our ecosystem can participate and allow the LLM models to have the best context and grounding possible. Today we lay the foundation for this vision by adding the [agent API](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.chatAgents2.d.ts#L8) in [a proposed state](https://code.visualstudio.com/api/advanced-topics/using-proposed-api).
+
+The agent API allows extensions to contribute agents that can answer specific questions by the user. Both the @workspace and the @vscode agent are implemented using this API. With agents, users can bring rich and up-to-date information from their inner and outer loop tools into AI conversations while staying in the editor flow. Agents are like experts for an area, and when a user explicitly mentions an @agent in their prompt, that prompt is forwarded to the extension that contributed that specific agent.
```

Comment:
@workspace and @vscode agents in this section are missing a backtick to nicely outline them